File: test/c/grid.c
/*****************************************************************************
*
* Rokko: Integrated Interface for libraries of eigenvalue decomposition
*
* Copyright (C) 2012-2015 by Rokko Developers https://github.com/t-sakashita/rokko
*
* Distributed under the Boost Software License, Version 1.0. (See accompanying
* file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
*
*****************************************************************************/

#include <rokko/rokko.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char** argv) {
  int i, n;
  int provided, ierr;
  struct rokko_grid grid;
  bool flag;
  int flag_int;
  
  ierr = MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);

  rokko_grid_construct(&grid, MPI_COMM_WORLD, rokko_grid_row_major);

  flag = rokko_grid_is_row_major(grid);
  printf("rokko_grid_is_row_major = %d\n", flag);
  flag = rokko_grid_is_col_major(grid);
  printf("rokko_grid_is_col_major = %d\n", flag);

  flag_int = rokko_grid_is_row_major(grid);
  printf("rokko_grid_is_row_major = %d\n", flag_int);
  flag_int = rokko_grid_is_col_major(grid);
  printf("rokko_grid_is_col_major = %d\n", flag_int);

  rokko_grid_destruct(&grid);

  MPI_Finalize();
}

File: test/python/grid/grid.py
#
# Rokko: Integrated Interface for libraries of eigenvalue decomposition
#
# Copyright (C) 2015-2019 by Rokko Developers https://github.com/t-sakashita/rokko
#
# Distributed under the Boost Software License, Version 1.0. (See accompanying
# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
#

import mpi4py
import numpy
import pyrokko

# This program must be run with 4 MPI processes.
grid_dim = 2

# row-major
g = pyrokko.grid(pyrokko.grid_row_major)
assert( g.nprocs == 4 )
grid_mat = numpy.arange(grid_dim**2, dtype='int').reshape(grid_dim, grid_dim)
assert( g.myrank == grid_mat[g.mine] )

# Defalut: row-major
g = pyrokko.grid()
assert( g.myrank == grid_mat[g.mine] )

# col-major
g = pyrokko.grid(pyrokko.grid_col_major)
grid_mat = numpy.arange(grid_dim**2, dtype='int').reshape(grid_dim, grid_dim, order='F')
assert( g.myrank == grid_mat[g.mine] )

mpi4py.MPI.Finalize()
